frontend/src/components/LoginPage.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { RootSchema } from "../store/root-schema";
+import { connect } from "react-redux";
+import { Dispatch } from "../store";
+import { connectWebSocket, setUsername } from "../store/actions";
+
+interface Props {
+  username: string;
+  connecting: boolean;
+  dispatch: Dispatch;
+}
+
+const LoginPage: React.FunctionComponent<Props> = ({
+  username,
+  connecting,
+  dispatch,
+}: Props) => {
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    if (connecting) return;
+
+    dispatch(setUsername(e.target.value));
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>): void => {
+    e.preventDefault();
+    dispatch(connectWebSocket(username));
```

Comment:
We should hide implementation so `connectWebSocket` can be renamed to `connect` `login` etc.